downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,422 @@
+  
+- ```ballerina
+  function filterEmployeesByDepartment(map<Employee> employees, string department) {
+    var result = from var e in employees // Compilation error now.
+        where e.department == department
+        select e.name;
+  }
+  ```
+
+- A bug that allowed ignoring possible completion with an error when using the `collect` clause in a query expression has been fixed.
+ 
+- ```ballerina
+  function calculateTotalSalary(stream<Employee, error?> strm, string dept) {
+    int total = from var {department, salary} in strm // Compilation error now.
+        where department == dept
+        collect sum(salary);
+  }
+  ```
+
+- A bug related to deciding the types of numeric literals has been fixed.
+
+    ```ballerina
+    2f|1 a = 2; // Compilation error now, `2` is considered to be `int`.
+    3d|6 b = 3; // Compilation error now, `3` is considered to be `int`.
+    3d|4f|6 b = 4; // Compilation error now, `4` is considered to be `int`.
+    ```
+
+- A bug that allowed an `anydata` value to be used in contexts expecting an `anydata & readonly` value has been fixed.
+
+    ```ballerina
+    public function main() {
+        anydata a = [];
+        anydata & readonly b = a; // Compilation error now.
+    }
+    ```
+  
+- A bug that allowed using object type inclusion with a `readonly` `service` object in a `readonly` non-`service` class has been fixed.
+
+    ```ballerina
+    public type A readonly & service object {
+        isolated remote function execute() returns int;
+    };
+  
+    readonly client class B {
+        *A; // Compilation error now.
+  
+        isolated remote function execute() returns int {
+            return 2;
+        }
+    }
+    ```
+
+- A bug that caused the runtime error messages to include the flattened representation of a union type instead of the name used in the type definition has been fixed.
+  
+    ```ballerina
+    type AC map<int>|string;
+    
+    public function main() returns error? {
+        map<anydata> x = {a: "Ballerina"};
+        AC|boolean _ = check x.cloneWithType();
+    }
+    ```
+  
+- A bug that resulted in an incorrect value when using a global variable, as a default value for the class field and function parameter has been fixed.
+    
+    ```ballerina
+    import ballerina/io;
+    
+    final int value = 100;
+    
+    class Foo {
+        int i = value;
+    }
+    
+    function bar(int k = value) returns int {
+        return k;
+    }
+    
+    public function main() {
+        Foo foo = new;
+        io:println(foo.i); // Prints `100` now.
+    }
+    ```
+
+    This now fails with `"'map<anydata>' value cannot be converted to '(AC|boolean)'"` instead of `"'map<anydata>' value cannot be converted to '(map<int>|string|boolean)'"`.
```

Comment:
This should come under the previous bullet point.

Reply:
Fixed.